pvanalytics/system.py

```diff
@@ -1,10 +1,289 @@
+    median_r2_min : float, default 0.9
+        Minimum :math:`r^2` for a curve fit to the median power or
+        irradiance at each minute of the day (Applies only if
+        `fit_median` is True).
+    fit_params : dict or None, default None
+        Minimum r-squared for curve fits according to the fraction of
+        data with clipping. This should be a dictionary with tuple
+        keys and dictionary values. The key must be a 2-tuple of
+        ``(clipping_min, clipping_max)`` where the values specify the
+        minimum and maximum fraction of data with clipping for which
+        the associated fit parameters are applicable. The values of
+        the dictionary are themselves dictionaries with keys
+        ``'fixed'`` and ``'tracking'``, which give the minimum
+        :math:`r^2` for the curve fits, and ``'fixed_max'`` which
+        gives the maximum :math:`r^2` for a quadratic fit if the
+        system appears to have a tracker.
+
+        If None :py:data:`PVFLEETS_FIT_PARAMS` is used.
+    seasonal_split : dict or str or None, default 'north-america'
+        A dictionary with two keys, 'winter' and 'summer' with a list of
+        integers specifying the winter months and summer months respectively.
+        Seasonal grouping can be disabled by passing `seasonal_split=None`.
+        Either season can be ignored by passing a dict that omits the key
+        or sets its value to None. The default value, 'north-america' uses
+        ``{'winter': [11, 12, 1, 2], 'summer': [5, 6, 7, 8]}`` which
+        works well for PV systems located in North America.
+
+    Returns
+    -------
+    Tracker
+        The tracking determined by curve fitting (FIXED, TRACKING, or
+        UNKNOWN).
+
+    Notes
+    -----
+    Derived from the PVFleets QA Analysis project.
+
+    See Also
+    --------
+
+    pvanalytics.features.orientation.tracking_nrel
+
+    pvanalytics.features.orientation.fixed_nrel
+
+    """
+    fit_params = fit_params or PVFLEETS_FIT_PARAMS
+    if seasonal_split == 'north-america':
+        seasonal_split = {'summer': [5, 6, 7, 8], 'winter': [11, 12, 1, 2]}
+    series_daytime = series[daytime]
+    clip_fraction = (clipping[daytime].sum() / len(clipping[daytime]))
+    if clip_fraction > clip_max:
+        return Tracker.UNKNOWN
+    bounds = _get_bounds(clip_fraction, fit_params)
+    if not seasonal_split:
+        return _infer_tracking(
+            series_daytime,
+            envelope_quantile,
+            fit_median,
+            median_r2_min,
+            bounds,
+            envelope_min_fraction,
+            median_min_fraction
+        )
+    summer = series_daytime[
+        series_daytime.index.month.isin(seasonal_split.get('summer') or [])
+    ]
+    winter = series_daytime[
+        series_daytime.index.month.isin(seasonal_split.get('winter') or [])
+    ]
+    if len(winter) == 0 and len(summer) == 0:
+        warnings.warn("no data in both winter and summer months")
```

Comment:
```suggestion
        warnings.warn("no data in winter or summer months")
```

Reply:
I thought this could be misleading, since the warning is only issued if both seasons are empty.

Reply:
'both winter and summer' made me think the code was expecting a subset of the data to be in both months.

Reply:
maybe "data is required in both winter and summer months"?

Reply:
"unknown tracking - both winter and summer months have no data"?

Reply:
better - I'm reacting to the g[rammar error](https://dictionary.cambridge.org/us/grammar/british-grammar/neither-neither-nor-and-not-either_2) when using "...both... not..." 

"Unknown tracking - neither winter nor summer months have data"

Reply:
```suggestion
        warnings.warn("Unknown tracking - neither winter nor summer months have data.")
```